src/scheduler/node_partition.cpp

```diff
@@ -919,10 +898,10 @@ find_alloc_np_cache(status *policy, np_cache ***pnpc_arr,
 			npc = new_np_cache();
 			if (npc != NULL) {
 				npc->ninfo_arr = ninfo_arr;
-				npc->resnames = dup_string_arr(const_cast<char **>(resnames));
+				npc->resnames = resnames;
 				npc->num_parts = num_parts;
 				npc->nodepart = nodepart;
-				if (npc->resnames == NULL || add_np_cache(pnpc_arr, npc) ==0) {
+				if (npc->resnames.empty() || add_np_cache(pnpc_arr, npc) ==0) {
```

Comment:
I'd remove the npc->resnames.empty().  If the memory failed to be allocated, an exception would have been thrown.  Also, add a space after the '=='